src/v/cloud_storage_clients/s3_client.cc

```diff
@@ -861,12 +861,13 @@ ss::future<> s3_client::do_delete_object(
       });
 }
 
-static std::variant<client::delete_objects_result, rest_error_response>
+std::variant<client::delete_objects_result, rest_error_response>
 iobuf_to_delete_objects_result(iobuf&& buf) {
     auto root = util::iobuf_to_ptree(std::move(buf), s3_log);
     auto result = client::delete_objects_result{};
     try {
-        if (root.count("Error.Code") > 0) {
+        if (auto error_code = root.get_optional<ss::sstring>("Error.Code");
```

Comment:
I don't expect you to change anything here, but (I believe) we can save some CPU instructions by doing extracting the child object here 
```cpp
if( auto error_payload = root.get_child_optional("Error"); error_payload.has_value() {
...
auto code = error_payload.get<ss::sstring>("Code");
}
```

changes:
* remove tree walking for each property
* remove the empty string optional and let it throw to show the expectations and intentions
* * these properties must always be present, if they are not we are likely doing something wrong or parsing a bogus message